wagtail/admin/active_filters.py

```diff
@@ -0,0 +1,208 @@
+from collections import namedtuple
+
+import django_filters
+from django.core.exceptions import ImproperlyConfigured
+from django.forms import BoundField, ModelChoiceField
+from django.http import QueryDict
+from django.utils.formats import date_format
+
+from wagtail.utils.registry import ObjectTypeRegistry
+from wagtail.utils.utils import flatten_choices
+
+# Represents a django-filters filter that is currently in force on a listing queryset
+ActiveFilter = namedtuple(
+    "ActiveFilter", ["auto_id", "field_label", "value", "removed_filter_url"]
+)
+
+
+class BaseFilterAdapter:
+    def __init__(
+        self,
+        filter: django_filters.Filter,
+        bound_field: BoundField,
+        value,
+        base_url: str,
+        query_dict: QueryDict,
+    ):
+        self.filter = filter
+        self.bound_field = bound_field
```

Comment:
Maybe we should also add something like `self.name = bound_field.name` so we can just do `self.name` instead of `self.bound_field.name` in various places